eip_7594/src/lib.rs

```diff
@@ -343,6 +379,57 @@ fn kzg_commitment_inclusion_proof<P: Preset>(
     proof
 }
 
+/**
+ * Return the sample count if allowing failures.
+ *
+ * This helper demonstrates how to calculate the number of columns to query per slot when
+ * allowing given number of failures, assuming uniform random selection without replacement.
+*/
+pub fn get_extended_sample_count(allowed_failures: u64) -> u64 {
+    // check that `allowed_failures` within the accepted range [0 -> NUMBER_OF_COLUMNS // 2]
+    assert!((0..(NumberOfColumns::U64 / 2)).contains(&allowed_failures));
+
+    // missing chunks for more than a half is the worst case
+    let worst_case_missing = NumberOfColumns::U64 / 2 + 1;
+
+    // modified from [math_lib](https://docs.rs/math_l/latest/src/math_l/math.rs.html#32-38) with compatible types
+    let math_comb = |n: u64, k: u64| -> f64 {
+        if k > n {
+            0_f64
+        } else {
+            (1..=k).fold(1_f64, |acc, i| acc * (n - i + 1) as f64 / i as f64)
+        }
+    };
+    
+    let hypergeom_cdf = |k: u64, m: u64, n: u64, big_n: u64| -> f64 {
+        (0..=k).fold(0_f64, |acc, i| acc + math_comb(n, i) * math_comb(m - n, big_n - i) / math_comb(m, big_n))
+    };
```

Comment:
The `math_comb` and `hypergeom_cdf` functions should be extracted out of the `get_extended_sample_count` function. Moving these to a separate module or as standalone functions will improve modularity, reusability, and testability. It will also enhance readability by separating core mathematical logic from the main function's workflow.

Reply:
good point, but i don't see it being else where. 

if we need in the future, we can seperate at that time. for now, I think it's okay as it is